src/test/core/Config_test.cpp

```diff
@@ -422,9 +433,11 @@ port_wss_admin
         {
             // read from file absolute path
             auto const cwd = current_path ();
+            detail::ConfigGuard g0(*this, "test_db");
             path const dataDirRel ("test_data_dir");
-            path const dataDirAbs (cwd / path ("test_db") / dataDirRel);
-            detail::RippledCfgGuard g (*this, "test_db", dataDirAbs.string (), "");
+            path const dataDirAbs(cwd / g0.subdir () / dataDirRel);
+            detail::RippledCfgGuard g (*this, g0.subdir().string(),
+                dataDirAbs.string (), "", false);
```

Comment:
Changing `subDir` in the `ConfigGuard`/`RippledCfgGuard`/`ValidatorsTxtGuard` constructors to a `path` would save you from having to call `.string()` on all these subdirectory paths.
(Same with `RippledCfgGuard`'s `dbPath`.)